src/market/market_token.cairo

```diff
@@ -68,11 +71,15 @@ mod MarketToken {
     }
 
     #[constructor]
-    fn constructor(ref self: ContractState, role_store_address: ContractAddress) {
+    fn constructor(
+        ref self: ContractState,
+        role_store_address: ContractAddress,
+        data_store_address: ContractAddress
+    ) {
         self.initializer(NAME, SYMBOL);
-        //Might need to inherit bank. 
-        // let mut bank: Bank::ContractState = Bank::unsafe_new_contract_state();
-        // IBank::initialize(ref bank, data_store_address, role_store_address)
+
+        let mut bank: Bank::ContractState = Bank::unsafe_new_contract_state();
+        IBank::initialize(ref bank, data_store_address, role_store_address);
         self.role_store.write(IRoleStoreDispatcher { contract_address: role_store_address });
```

Comment:
what do you mean? MarketToken is inheriting Bank contract. How are we going to use it here?

Reply:
Since we have `market_token <- bank <- role_module` and `role_store` is already stored in role_module maybe we can remove `role_store` from `market_token` storage and do something like:
```
let mut role_module: RoleModule::ContractState =
                RoleModule::unsafe_new_contract_state();
role_module.only_controller();
```
where we need to check roles. Thus we avoid storing the same variable two times.

Reply:
ok. I removed it from storage. Is it okay like this?
https://github.com/tevrat-aksoy/satoru/commit/9f1f5c66d181a51bcf03208f590cd34c3dbbd139

Reply:
Perfect !